.github/workflows/format.yml

```
@@ -0,0 +1,37 @@
+name: 'Format Code'
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  format:
+    runs-on: ubuntu-latest
+    env:
+      NODE_OPTIONS: "--max_old_space_size=4096"
+    steps:
+      - name: Check out code using Git
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+          # Needs access to push to main
+          token: ${{ secrets.FREDKBOT_GITHUB_TOKEN }}
+      - name: Setup PNPM
+        uses: pnpm/action-setup@v2.2.1
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'pnpm'
+      - name: Install dependencies
+        run: pnpm install
```

We have a local action for these steps that we use to keep it consistent between workflows:

```suggestion
      - name: Install Tools & Dependencies
        uses: ./.github/actions/install
```